src/temp.bash

```diff
@@ -99,7 +99,13 @@ temp_make() {
   template+='-XXXXXXXXXX'
 
   local path
-  path="$(mktemp --directory --tmpdir="$BATS_TMPDIR" -- "$template" 2>&1)"
+
+    if [ "$(uname -s)" == "Darwin" ]; then
```

Comment:
Wouldn't you want to detect which mktemp rather than the OS? 

```bash
if mktemp --version >/dev/null 2>&1 ; then
    echo Using GNU mktemp
else
    echo Using BSD mktemp
fi
```

Reply:
Feel free to add my remote and implement that change, I currently don't have `bats` set up on my machine, and given that this PR is been ignored for about 3 months now, I don't think it's worth the effort to try to improve it.

Reply:
I've verified this works as expected on my mac.

Reply:
Given mac doesn't work at all, could we land this rather than bike shed it?